packages/zowe-explorer/src/trees/uss/ZoweUSSNode.ts

```diff
@@ -393,8 +393,24 @@ export class ZoweUSSNode extends ZoweTreeNode implements IZoweUSSTreeNode {
         try {
             await vscode.workspace.fs.rename(oldUri, newUri, { overwrite: false });
         } catch (err) {
-            Gui.errorMessage(err.message);
-            return;
+            if (err instanceof vscode.FileSystemError && err.code === "FileExists") {
+                try {
+                    const fileList = await UssFSProvider.instance.listFiles(this.getProfile(), newUri, true);
+                    if (!fileList.success) {
+                        Gui.errorMessage(err.message);
+                        return;
+                    }
+                } catch (err) {
+                    if ((err as any).name === "ImperativeError" && Number(err.errorCode) === imperative.RestConstants.HTTP_STATUS_404) {
+                        const parent = UssFSProvider.instance._lookupParentDirectory(newUri);
+                        parent.entries.delete(path.posix.basename(newUri.path));
+                        await vscode.workspace.fs.rename(oldUri, newUri);
+                    }
+                }
```

Comment:
Would it make sense to handle this error inside the `UssFSProvider.rename` function?

This would allow `_lookupParentDirectory` to be called without needing to make it a public method.

Not necessarily suggesting a change, if this error handling is only needed in the ZE tree and not applicable to every `zowe-uss` rename operation.